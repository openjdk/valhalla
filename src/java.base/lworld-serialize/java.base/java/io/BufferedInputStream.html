<!DOCTYPE html>
<html><head><meta charset="UTF-8">
<title>class java.base/java.io.BufferedInputStream</title>
<meta name="generator" content="apidiff">
<link rel="stylesheet" type="text/css; charset=UTF-8" href="../../../resources/apidiff.css">
</head>
<body class="type"><header><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</header>
<main><div class="pageHeading"><div><span class="label">Module</span>&nbsp;<a href="../../../java.base/module-summary.html">java.base</a></div>
<div><span class="label">Package</span>&nbsp;<a href="../../../java.base/java/io/package-summary.html">java.io</a></div>
<h1>Class&nbsp;BufferedInputStream</h1>
</div>
<div class="element"><span class="diff">&cross;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">class</span> BufferedInputStream <div class="superclass"><span class="keyword">extends</span> java.io.FilterInputStream</div>
</div>
</div>
<div class="apiDescriptions"></div>
<section class="enclosed"><h2>Fields</h2>
<ul><li><div class="element" id="buf"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span> <span class="keyword">volatile</span></span> <span><span class="keyword">byte</span>[]</span> buf</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="count"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span class="keyword">int</span> count</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="marklimit"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span class="keyword">int</span> marklimit</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="markpos"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span class="keyword">int</span> markpos</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="pos"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span class="keyword">int</span> pos</div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="enclosed"><h2>Constructors</h2>
<ul><li><div class="element" id="&lt;init&gt;(InputStream)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> BufferedInputStream(<span class="parameters"><span class="parameter">java.io.InputStream arg0</span></span>)</div>
</div>
</li><li><div class="element" id="&lt;init&gt;(InputStream,int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> BufferedInputStream(<span class="parameters"><span class="parameter">java.io.InputStream arg0</span>, <span class="parameter"><span class="keyword">int</span> arg1</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="enclosed"><h2>Methods</h2>
<ul><li><div class="element" id="available()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">int</span> available() <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="close()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> close() <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="mark(int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> mark(<span class="parameters"><span class="parameter"><span class="keyword">int</span> arg0</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="markSupported()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">boolean</span> markSupported()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="read()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">int</span> read() <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
</div>
</li><li><div class="element" id="read(byte[],int,int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">int</span> read(<span class="parameters"><span class="parameter"><span><span class="keyword">byte</span>[]</span> arg0</span>, <span class="parameter"><span class="keyword">int</span> arg1</span>, <span class="parameter"><span class="keyword">int</span> arg2</span></span>) <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="reset()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> reset() <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="skip(long)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">long</span> skip(<span class="parameters"><span class="parameter"><span class="keyword">long</span> arg0</span></span>) <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="transferTo(OutputStream)"><span class="single">&#x2460;</span><span class="missing">Only in: jdk-git; not in: lworld.</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">long</span> transferTo(<span class="parameters"><span class="parameter">java.io.OutputStream arg0</span></span>) <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
</div>
</li></ul>
</section>
<section class="summary"><h2>Summary</h2>
<table class="summary"><caption>Differences</caption>
<thead><tr><td rowspan="2"></td>
<th colspan="3">Elements</th>
<th class="allZero" colspan="3">Comments</th>
<th class="allZero" colspan="3">Descriptions</th>
<th rowspan="2">Total</th>
</tr>
<tr><th class="allZero">Added</th>
<th class="allZero">Changed</th>
<th>Removed</th>
<th class="allZero">Added</th>
<th class="allZero">Changed</th>
<th class="allZero">Removed</th>
<th class="allZero">Added</th>
<th class="allZero">Changed</th>
<th class="allZero">Removed</th>
</tr>
</thead>
<tbody><tr><th scope="row"><a href="../../../java.base/java/io/BufferedInputStream.html#transferTo(OutputStream)">transferTo(OutputStream)</a></th>
<td></td>
<td></td>
<td>1</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>1</td>
</tr>
</tbody>
<tfoot><tr><th scope="row">Total</th>
<td></td>
<td></td>
<td>1</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>1</td>
</tr>
</tfoot>
</table>
</section>
</main>
<footer><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</footer>
</body>
</html>
