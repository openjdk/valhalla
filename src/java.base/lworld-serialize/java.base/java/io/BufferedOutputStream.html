<!DOCTYPE html>
<html><head><meta charset="UTF-8">
<title>class java.base/java.io.BufferedOutputStream</title>
<meta name="generator" content="apidiff">
<link rel="stylesheet" type="text/css; charset=UTF-8" href="../../../resources/apidiff.css">
</head>
<body class="type"><header><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</header>
<main><div class="pageHeading"><div><span class="label">Module</span>&nbsp;<a href="../../../java.base/module-summary.html">java.base</a></div>
<div><span class="label">Package</span>&nbsp;<a href="../../../java.base/java/io/package-summary.html">java.io</a></div>
<h1>Class&nbsp;BufferedOutputStream</h1>
</div>
<div class="element"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">class</span> BufferedOutputStream <div class="superclass"><span class="keyword">extends</span> java.io.FilterOutputStream</div>
</div>
</div>
<div class="apiDescriptions"></div>
<section class="enclosed"><h2>Fields</h2>
<ul><li><div class="element" id="buf"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span><span class="keyword">byte</span>[]</span> buf</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="count"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span class="keyword">int</span> count</div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="enclosed"><h2>Constructors</h2>
<ul><li><div class="element" id="&lt;init&gt;(OutputStream)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> BufferedOutputStream(<span class="parameters"><span class="parameter">java.io.OutputStream arg0</span></span>)</div>
</div>
</li><li><div class="element" id="&lt;init&gt;(OutputStream,int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> BufferedOutputStream(<span class="parameters"><span class="parameter">java.io.OutputStream arg0</span>, <span class="parameter"><span class="keyword">int</span> arg1</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="enclosed"><h2>Methods</h2>
<ul><li><div class="element" id="flush()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> flush() <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="write(int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> write(<span class="parameters"><span class="parameter"><span class="keyword">int</span> arg0</span></span>) <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
</div>
</li><li><div class="element" id="write(byte[],int,int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> write(<span class="parameters"><span class="parameter"><span><span class="keyword">byte</span>[]</span> arg0</span>, <span class="parameter"><span class="keyword">int</span> arg1</span>, <span class="parameter"><span class="keyword">int</span> arg2</span></span>) <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="summary"><h2>Summary</h2>
No differences.</section>
</main>
<footer><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</footer>
</body>
</html>
