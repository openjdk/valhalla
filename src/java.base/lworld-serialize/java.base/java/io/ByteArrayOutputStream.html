<!DOCTYPE html>
<html><head><meta charset="UTF-8">
<title>class java.base/java.io.ByteArrayOutputStream</title>
<meta name="generator" content="apidiff">
<link rel="stylesheet" type="text/css; charset=UTF-8" href="../../../resources/apidiff.css">
</head>
<body class="type"><header><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</header>
<main><div class="pageHeading"><div><span class="label">Module</span>&nbsp;<a href="../../../java.base/module-summary.html">java.base</a></div>
<div><span class="label">Package</span>&nbsp;<a href="../../../java.base/java/io/package-summary.html">java.io</a></div>
<h1>Class&nbsp;ByteArrayOutputStream</h1>
</div>
<div class="element"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">class</span> ByteArrayOutputStream <div class="superclass"><span class="keyword">extends</span> java.io.OutputStream</div>
</div>
</div>
<div class="apiDescriptions"></div>
<section class="enclosed"><h2>Fields</h2>
<ul><li><div class="element" id="buf"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span><span class="keyword">byte</span>[]</span> buf</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="count"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">protected</span></span> <span class="keyword">int</span> count</div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="enclosed"><h2>Constructors</h2>
<ul><li><div class="element" id="&lt;init&gt;()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> ByteArrayOutputStream()</div>
</div>
</li><li><div class="element" id="&lt;init&gt;(int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> ByteArrayOutputStream(<span class="parameters"><span class="parameter"><span class="keyword">int</span> arg0</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="enclosed"><h2>Methods</h2>
<ul><li><div class="element" id="close()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> close() <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="reset()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> reset()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="size()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">int</span> size()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="toByteArray()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span><span class="keyword">byte</span>[]</span> toByteArray()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="toString()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.lang.String toString()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="toString(int)"><span class="same">&check;</span><div class="signature"><span class="annotation">@Deprecated</span><br>
<span class="modifiers"><span class="keyword">public</span></span> java.lang.String toString(<span class="parameters"><span class="parameter"><span class="keyword">int</span> arg0</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="toString(String)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.lang.String toString(<span class="parameters"><span class="parameter">java.lang.String arg0</span></span>) <span class="throws"><span class="keyword">throws</span> java.io.UnsupportedEncodingException</span></div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="toString(Charset)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.lang.String toString(<span class="parameters"><span class="parameter">java.nio.charset.Charset arg0</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="write(int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> write(<span class="parameters"><span class="parameter"><span class="keyword">int</span> arg0</span></span>)</div>
</div>
</li><li><div class="element" id="write(byte[],int,int)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> write(<span class="parameters"><span class="parameter"><span><span class="keyword">byte</span>[]</span> arg0</span>, <span class="parameter"><span class="keyword">int</span> arg1</span>, <span class="parameter"><span class="keyword">int</span> arg2</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="writeBytes(byte[])"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> writeBytes(<span class="parameters"><span class="parameter"><span><span class="keyword">byte</span>[]</span> arg0</span></span>)</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="writeTo(OutputStream)"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> writeTo(<span class="parameters"><span class="parameter">java.io.OutputStream arg0</span></span>) <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="summary"><h2>Summary</h2>
No differences.</section>
</main>
<footer><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</footer>
</body>
</html>
