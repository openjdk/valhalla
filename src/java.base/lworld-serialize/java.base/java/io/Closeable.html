<!DOCTYPE html>
<html><head><meta charset="UTF-8">
<title>interface java.base/java.io.Closeable</title>
<meta name="generator" content="apidiff">
<link rel="stylesheet" type="text/css; charset=UTF-8" href="../../../resources/apidiff.css">
</head>
<body class="type"><header><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</header>
<main><div class="pageHeading"><div><span class="label">Module</span>&nbsp;<a href="../../../java.base/module-summary.html">java.base</a></div>
<div><span class="label">Package</span>&nbsp;<a href="../../../java.base/java/io/package-summary.html">java.io</a></div>
<h1>Interface&nbsp;Closeable</h1>
</div>
<div class="element"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span> <span class="keyword">abstract</span></span> <span class="keyword">interface</span> Closeable <div class="superinterfaces"><span class="keyword">extends</span> java.lang.AutoCloseable</div>
</div>
</div>
<div class="apiDescriptions"></div>
<section class="enclosed"><h2>Methods</h2>
<ul><li><div class="element" id="close()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span> <span class="keyword">abstract</span></span> <span class="keyword">void</span> close() <span class="throws"><span class="keyword">throws</span> java.io.IOException</span></div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="summary"><h2>Summary</h2>
No differences.</section>
</main>
<footer><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</footer>
</body>
</html>
