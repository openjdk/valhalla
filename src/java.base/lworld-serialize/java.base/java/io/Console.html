<!DOCTYPE html>
<html><head><meta charset="UTF-8">
<title>class java.base/java.io.Console</title>
<meta name="generator" content="apidiff">
<link rel="stylesheet" type="text/css; charset=UTF-8" href="../../../resources/apidiff.css">
</head>
<body class="type"><header><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</header>
<main><div class="pageHeading"><div><span class="label">Module</span>&nbsp;<a href="../../../java.base/module-summary.html">java.base</a></div>
<div><span class="label">Package</span>&nbsp;<a href="../../../java.base/java/io/package-summary.html">java.io</a></div>
<h1>Class&nbsp;Console</h1>
</div>
<div class="element"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span> <span class="keyword">final</span></span> <span class="keyword">class</span> Console <div class="superclass"><span class="keyword">extends</span> java.lang.Object</div>
<div class="superinterfaces"><span class="keyword">implements</span> java.io.Flushable</div>
</div>
</div>
<div class="apiDescriptions"></div>
<section class="enclosed"><h2>Methods</h2>
<ul><li><div class="element" id="charset()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.nio.charset.Charset charset()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="flush()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span class="keyword">void</span> flush()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="format(String,Object[])"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.io.Console format(<span class="parameters"><span class="parameter">java.lang.String arg0</span>, <span class="parameter"><span>java.lang.Object[]</span> arg1</span></span>)</div>
</div>
</li><li><div class="element" id="printf(String,Object[])"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.io.Console printf(<span class="parameters"><span class="parameter">java.lang.String arg0</span>, <span class="parameter"><span>java.lang.Object[]</span> arg1</span></span>)</div>
</div>
</li><li><div class="element" id="readLine()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.lang.String readLine()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="readLine(String,Object[])"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.lang.String readLine(<span class="parameters"><span class="parameter">java.lang.String arg0</span>, <span class="parameter"><span>java.lang.Object[]</span> arg1</span></span>)</div>
</div>
</li><li><div class="element" id="readPassword()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span><span class="keyword">char</span>[]</span> readPassword()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="readPassword(String,Object[])"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> <span><span class="keyword">char</span>[]</span> readPassword(<span class="parameters"><span class="parameter">java.lang.String arg0</span>, <span class="parameter"><span>java.lang.Object[]</span> arg1</span></span>)</div>
</div>
</li><li><div class="element" id="reader()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.io.Reader reader()</div>
<div class="apiDescriptions"></div>
</div>
</li><li><div class="element" id="writer()"><span class="same">&check;</span><div class="signature"><span class="modifiers"><span class="keyword">public</span></span> java.io.PrintWriter writer()</div>
<div class="apiDescriptions"></div>
</div>
</li></ul>
</section>
<section class="summary"><h2>Summary</h2>
No differences.</section>
</main>
<footer><div class="bar"><div class="info">jdk-git : lworld</div>
<nav><ul><li><a href="../../../index.html">Overview</a></li></ul>
</nav>
</div>
</footer>
</body>
</html>
