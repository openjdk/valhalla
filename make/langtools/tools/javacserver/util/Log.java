/*
 * Copyright (c) 2012, 2022, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

package javacserver.util;

import java.io.PrintWriter;
import java.io.StringWriter;
import java.io.Writer;

/**
 * Utility class only for javacserver logging.
 *
 * Logging in javacserver has special requirements when running in server/client
 * mode. Most of the log messages is generated server-side, but the server
 * is typically spawned by the client in the background, so the user usually
 * does not see the server stdout/stderr. For this reason log messages needs
 * to relayed back to the client that performed the request that generated the
 * log message. To support this use case this class maintains a per-thread log
 * instance so that each connected client can have its own instance that
 * relays messages back to the requesting client.
 *
 * On the client-side there will typically just be one Log instance.
 */
public class Log {
    public enum Level {
        ERROR,
        WARN,
        INFO,
        DEBUG,
        TRACE;
    }

    private static Log stdOutErr = new Log(new PrintWriter(System.out), new PrintWriter(System.err));
    private static ThreadLocal<Log> logger = new ThreadLocal<>();

    protected PrintWriter err; // Used for error and warning messages
    protected PrintWriter out; // Used for other messages
    protected Level level = Level.INFO;

    public Log(Writer out, Writer err) {
        this.out = out == null ? null : new PrintWriter(out, true);
        this.err = err == null ? null : new PrintWriter(err, true);
    }

    public static void setLogForCurrentThread(Log log) {
        logger.set(log);
    }

    public static void setLogLevel(Level l) {
        get().level = l;
    }

    public static void debug(String msg) {
        log(Level.DEBUG, msg);
    }

    public static void debug(Throwable t) {
        log(Level.DEBUG, t);
    }

    public static void error(String msg) {
        log(Level.ERROR, msg);
    }

    public static void error(Throwable t) {
        log(Level.ERROR, t);
    }

    public static void log(Level l, String msg) {
        get().printLogMsg(l, msg);
    }

    public static void log(Level l, Throwable t) {
        StringWriter sw = new StringWriter();
        t.printStackTrace(new PrintWriter(sw, true));
        log(l, sw.toString());
    }

    public static boolean isDebugging() {
        return get().isLevelLogged(Level.DEBUG);
    }

    protected boolean isLevelLogged(Level l) {
        return l.ordinal() <= level.ordinal();
    }

    public static Log get() {
        Log log = logger.get();
        return log != null ? log : stdOutErr;
    }

    protected void printLogMsg(Level msgLevel, String msg) {
        if (isLevelLogged(msgLevel)) {
            PrintWriter pw = msgLevel.ordinal() <= Level.WARN.ordinal() ? err : out;
            pw.println(msg);
        }
    }
}
