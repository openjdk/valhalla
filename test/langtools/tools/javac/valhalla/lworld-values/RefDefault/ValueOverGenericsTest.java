/*
 * @test /nodynamiccopyright/
 * @summary Value types cannot parameterize generic types (except under experimental mode)
 * @compile/fail/ref=ValueOverGenericsTest.out -XDrawDiagnostics ValueOverGenericsTest.java
 *
 */

import java.util.ArrayList;
import java.io.Serializable;

primitive class ValueOverGenericsTest.val {
    int x = 10;
    ArrayList<ValueOverGenericsTest.val> ax = null;
    void foo(ArrayList<? extends ValueOverGenericsTest.val> p) {
        new <ValueOverGenericsTest.val> ArrayList<Object>();
        this.<ValueOverGenericsTest.val>foo(null);
        Object o = (ValueOverGenericsTest.val & Serializable) null;
    }
}
