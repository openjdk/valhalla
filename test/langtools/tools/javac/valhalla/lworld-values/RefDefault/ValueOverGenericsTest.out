ValueOverGenericsTest.java:13:36: compiler.err.type.found.req: ValueOverGenericsTest.val, (compiler.misc.type.req.ref)
ValueOverGenericsTest.java:14:24: compiler.err.type.found.req: ValueOverGenericsTest.val, (compiler.misc.type.req.ref)
ValueOverGenericsTest.java:15:35: compiler.err.type.found.req: ValueOverGenericsTest.val, (compiler.misc.type.req.ref)
ValueOverGenericsTest.java:16:36: compiler.err.type.found.req: ValueOverGenericsTest.val, (compiler.misc.type.req.ref)
ValueOverGenericsTest.java:17:42: compiler.err.type.found.req: ValueOverGenericsTest.val, (compiler.misc.type.req.ref)
ValueOverGenericsTest.java:17:63: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: compiler.misc.type.null, ValueOverGenericsTest.val&java.io.Serializable)
6 errors
